src/tcgwars/logic/impl/gen8/BattleStyles.groovy

```diff
@@ -710,7 +680,7 @@ public enum BattleStyles implements LogicCardInfo {
           text "Attach up to 3 [R] Energy cards from your discard pile to your Pokémon in any way you like."
           energyCost COLORLESS
           attackRequirement {
-            assert discardPile.filterByEnergyType(R)
+            assert discardPile.filterByEnergyType(R) : "No $R Energy in your Discard Pile."
```

Comment:
This is just my personal preference, but I prefer [R] here to $R. [X] seems to be used almost everywhere else.

Reply:
I prefer using the string representation because FIRE is less confusing than R (I totally put $F the first time)